packages/cli/src/lib/version-check/update-message.js

```diff
@@ -0,0 +1,52 @@
+import { info } from 'log-symbols'
+import terminalLink from 'terminal-link'
+
+import { getVersions, initStorage, getHistory, setHistory } from './storage'
+
+export async function registerExitMessage() {
+  await initStorage()
+  const shouldLog = await logWithBackoff()
+  if (!shouldLog) return
+  // NOTE: could move this check to exit to avoid off by one
+  const storedVersion = await getVersions()
+  process.on('exit', function (exitCode) {
+    if (storedVersion !== undefined) {
+      logMessage(storedVersion, exitCode)
+    }
+  })
+}
+
+// use exponential backoff to determine if this rw run should show upgrade message
+// log - log - no - no - no - log - no - ...
```

Comment:
@geoff616 this is just 💯 and alleviates one of my primary concerns regarding user behavior + annoyance == users just turning things off. Just 🗿!

Reply:
This is amazing, thanks @geoff616  - I'm a bit reluctant to merge something like this without tests, especially with the exponential backoff.

Reply:
Thanks, will add tests this week

Reply:
@geoff616 Just hold off for a second, we're going to discuss internally, I would like to propose an alternative approach that may be a bit easier to implement and support.

Reply:
@geoff616 Quick update --> Peter and I didn't get a chance to connect today but will do so tomorrow.